app/js/scripts.js

```diff
@@ -137,6 +137,7 @@ function editReportById(reportId){
     changeReportView('computer', 'edit');
     $("#report-name-field").html(reportObject.name);
     $("#new-report-name").val(reportObject.name);
+    $("#existing-report-id").val(reportId)
```

Comment:
This wasn't happening before and is required to update the report. It was thinking that it was a new report. That's why it was duplicating the report in the DB when it worked.